realm/realm-library/src/main/cpp/io_realm_internal_objectstore_OsRemoteMongoCollection.cpp

```diff
@@ -58,9 +81,110 @@ Java_io_realm_internal_objectstore_OsRemoteMongoCollection_nativeCount(JNIEnv* e
                                                                        jobject j_callback) {
     try {
         RemoteMongoCollection* collection = reinterpret_cast<RemoteMongoCollection*>(j_collection_ptr);
-        bson::BsonDocument filter(JniBsonProtocol::jstring_to_bson(env, j_filter));
+        bson::BsonDocument bson_filter(JniBsonProtocol::jstring_to_bson(env, j_filter));
```

Comment:
We should probably add checks on the actual type even though we have quite good control on the input side in Java, as people could use a codec registry that encodes object unexpectedly. 
```
if (args.type() != bson::Bson::Type::Array) {
       throw util::invalid_argument("Function arguments must be a BSON array");
}
```
which will throw an `IllegalArgumentException` to Java. 

Then it should be safe to cast it with `static_cast<bson::BsonArray>(args)` to get the right type without hitting the assertions.

Reply:
We should probably do a `<T> JniBsonProtocol::checked_jstring_to_bson(...)`. To ease it, but my template programming is a bit rusty, so I cannot write from the top of my head. I can do it as part of the functions PR, then you can await the implementation and guard the arguments later.

Reply:
I will add a FIXME to remind me of that.